src/test/java/com/microsoft/sqlserver/jdbc/fedauth/FedauthTest.java

```diff
@@ -352,6 +355,18 @@ public void testAADServicePrincipalAuthWrong() {
         validateException(url, "R_BothUserPasswordandDeprecated");
     }
 
+    @Tag(Constants.xSQLv11)
+    @Tag(Constants.xSQLv12)
+    @Tag(Constants.xSQLv14)
+    @Tag(Constants.xAzureSQLDW)
+    @Tag(Constants.reqExternalSetup)

```

Comment:
why reqExternalSetup?

Reply:
@Tag(Constants.fedAuth)

Reply:
> why reqExternalSetup?

Tests need a clientId and secret in order to get a token through the callback used in the test.

>@tag(Constants.fedAuth)

The class is already tagged as fedauth so we don't need to do it at the class level.